examples/40_paper/2018_kdd_rijn_example.py

```diff
@@ -8,9 +8,116 @@
 ~~~~~~~~~~~
 
 | Hyperparameter importance across datasets
-| Jan van Rijn and Frank Hutter
+| Jan N. van Rijn and Frank Hutter
 | In *Proceedings of the 24th ACM SIGKDD International Conference on Knowledge Discovery & Data Mining*, 2018
 | Available at https://dl.acm.org/citation.cfm?id=3220058
-
-This is currently a placeholder.
 """
+import fanova
+import json
+import logging
+import matplotlib.pyplot as plt
+import openml
+import pandas as pd
+import seaborn as sns
+
+root = logging.getLogger()
+root.setLevel(logging.INFO)
+
+##############################################################################
+# With the advent of automated machine learning, automated hyperparameter
+# optimization methods are by now routinely used in data mining. However, this
+# progress is not yet matched by equal progress on automatic analyses that
+# yield information beyond performance-optimizing hyperparameter settings.
+# In this example, we aim to answer the following two questions: Given an
+# algorithm, what are generally its most important hyperparameters?
+#
+# This work is carried out on the OpenML-100 benchmark suite, which can be
+# obtained by `openml.study.get_suite('OpenML100')`. In this example, we
+# conduct the experiment on the Support Vector Machine (`flow_id=7707`)
+# with specific kernel (we will perform a post-process filter operation for
+# this). We should set some other experimental parameters (number of results
+# per task, evaluation measure and the number of trees of the internal
+# functional Anova) before the fun can begin.
+#
+# Note that we simplify the example in several ways
+# 1) We only consider numerical hyperparameters
+# 2) We consider all hyperparameters that are numerical (in reality, some
+#    hyperparameters might be inactive (e.g., `degree`) or irrelevant (e.g.,
+#    `random_state`)
+# 3) We assume all hyperparameters to be on uniform scale
+#
+# Any difference in conclusion between the actual paper and the presented
+# results is most likely due to one of these simplifications. For example,
+# # the hyperparameter C looks rather insignificant, whereas it is quite
+# important when it is put on a log-scale. All these simplifications can be
+# addressed by defining a ConfigSpace. For a more elaborated example that uses
+# this, please see:
+# https://github.com/janvanrijn/openml-pimp/blob/d0a14f3eb480f2a90008889f00041bdccc7b9265/examples/plot/plot_fanova_aggregates.py # noqa F401
+
+suite = openml.study.get_suite('OpenML100')
+flow_id = 7707
+parameter_filters = {
+    'sklearn.svm.classes.SVC(17)_kernel': 'sigmoid'
+}
+evaluation_measure = 'predictive_accuracy'
+limit_per_task = 500
+n_trees = 16
+
+fanova_results = []
+# we will obtain all results from OpenML per task. Practice has shown that this places the bottleneck on the
+# communication with OpenML, and for iterated experimenting it is better to cache the results in a local file.
```

Comment:
Is there any way for speeding this up? Otherwise, would it be possible to only conduct a part of the evaluation to limit the running time of this example?